src/Symfony/Component/RateLimiter/Policy/SlidingWindow.php

```diff
@@ -124,4 +104,25 @@ public function getRetryAfter(): \DateTimeImmutable
     {
         return \DateTimeImmutable::createFromFormat('U.u', sprintf('%.6F', $this->windowEndAt));
     }
+
+    /**
+     * @internal
+     */
+    public function __serialize(): array
+    {
+        return [
+            pack('NNN', $this->hitCount, $this->hitCountForLastWindow, $this->intervalInSeconds).$this->id => $this->windowEndAt,
```

Comment:
Thanks for the updates @nicolas-grekas - may I suggest doing the same for the other types of RateLimiter\Policy\* ? I assume they suffer from the same overly-large-payload problem. I am not super familiar with pack but I'll have a more detailed look at the change you did later this week.

Reply:
done